<launch>
<!-- This was copied from the lane_following.launch file. -->
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
	<arg name="ai_trafo_mode" default="cb" doc="'cb' for colo balance only; 'both' for color balance and linear trafo"/>
	<arg name="ai_interval" default="5" doc="interval with which the linear trafo gets updated. color balance is performed every second."/>
	<arg name="verbose" default="false"/>

	<!-- start basic args -->
	<include file="$(find duckietown_demos)/launch/master.launch">
		<arg name="veh" value="$(arg veh)"/>
		<!--arg name="verbose" value="$(arg verbose)" /-->
    	<arg name="/camera/raw" default="true" />
		<arg name="param_file_name" default="default" />
		<arg name="line_detector_param_file_name" default="default" />
		<arg name="anti_instagram" default="true" />
		<arg name="ai_trafo_mode" value="$(arg ai_trafo_mode)"/>
		<arg name="ai_interval" value="$(arg ai_interval)"/>
		<arg name="/lane_following/stop_line_filter" default="false" /> <!-- will not every allow a transition into COORDINATION mode -->
		<arg name="vehicle_avoidance" default="false"/>
		<!-- ADDED: turn off the lane_controller node. -->
		<arg name="/lane_following/lane_controller" default="false"/>
		<!-- ADDED: turn off the duckiebot_vizualizer node. -->
		<arg name="visualization" default="false"/>
	</include>
	
	<!-- My Pure Pursuit Lane controller -->
	<group>
		<remap from="pure_pursuit_controller_node/lane_pose" to="lane_filter_node/lane_pose"/>
		<!-- <remap from="pure_pursuit_controller_node/seglist_filtered" to="ground_projection/lineseglist_out"/> -->
		<remap from="pure_pursuit_controller_node/follow_point" to="duckiebot_visualizer/follow_point"/>
		<remap from="pure_pursuit_controller_node/path_points" to="duckiebot_visualizer/path_points"/>
		<remap from="pure_pursuit_controller_node/seglist_filtered" to="lane_filter_node/seglist_filtered"/>
		<remap from="pure_pursuit_controller_node/car_cmd" to="car_cmd_switch_node/cmd"/>
		<!-- <remap from="pure_pursuit_controller_node/car_cmd" to="lane_controller_node/car_cmd"/> -->

		<include file="$(find pure_pursuit)/launch/pure_pursuit_controller_node.launch">
			<arg name="veh" value="$(arg veh)"/>
			<!-- TODO: use this to pass a specific parameter file. (not needed here.) -->
			<arg name="param_file_name" value="brigitte"/>
		</include>
	</group>

	<!-- Local Duckiebot Visualizer -->
	<group>
		<remap from="duckiebot_visualizer/segment_list" to="ground_projection/lineseglist_out"/>
		<remap from="duckiebot_visualizer/segment_list_filtered" to="lane_filter_node/seglist_filtered"/>
		<include file="$(find duckiebot_visualizer_local)/launch/duckiebot_visualizer.launch">
			<arg name="veh" value="$(arg veh)" />
		</include>
	</group>

	<!-- Vehicle detection node -->
	<group>
		<remap from="/default/vehicle_detection_node/image" to="camera_node/image/compressed"/>
		<include file="$(find pure_pursuit)/launch/vehicle_detection.launch">
			<arg name="veh" value="$(arg veh)" />
		</include>
	</group>
</launch>
